### PR TITLE
[Core] Add opt-in Inductor compute/communication overlap for diffusion models

### PR DESCRIPTION
**Pull request by kTorp**
## Purpose

This PR adds an opt-in `--diffusion-compile-reorder-comm-overlap` setting for vLLM-Omni. When enabled, generic diffusion compilation passes Inductor options that reorder communication and computation to improve overlap and reduce latency for distributed diffusion workloads.

The setting is disabled by default and applies to both regional and full compilation. It has no effect with `--enforce-eager` or pipelines that provide their own compilation setup.

## Test Plan

- [x] Before and after results, with commands, shown below.

## Test Result

Besides earlier results shown below the branch has also been updated and new tests were run on B200 8xGPUs showing similar improvements without quality loss and stable peak VRAM. Results below were taken on an MI300 8xGPU.

Wan2.2
```
vllm-omni serve Wan-AI/Wan2.2-T2V-A14B-Diffusers \
  --omni \
  --port 8098 \
  --ulysses-degree 8
  
curl -X POST http://localhost:8098/v1/videos/sync \
  -F 'prompt=Summer beach vacation style, a white cat wearing sunglasses sits on a surfboard. The fluffy-furred feline gazes directly at the camera with a relaxed expression. Blurred beach scenery forms the background featuring crystal-clear waters, distant green hills, and a blue sky dotted with white clouds. The cat assumes a naturally relaxed posture, as if savoring the sea breeze and warm sunlight. A close-up shot highlights the felines intricate details and the refreshing atmosphere of the seaside.' \
  -F size=1280x720 \
  -F num_frames=81 \
  -F num_inference_steps=40 \
  -F guidance_scale=5.0 \
  -F seed=42 \
  -o iter_1.mp4 \
  -D -
```

Qwen-Image-Edit
```
VLLM_LOGGING_LEVEL=DEBUG vllm serve Qwen/Qwen-Image-Edit --omni --port 8098 --ulysses-degree 8

curl -s -X POST http://localhost:8098/v1/images/edits \
  -F "model=Qwen/Qwen-Image-Edit" \
  -F "image=@/app/data/flux_cat.png" \
  -F "prompt=Add a cool hat to the cat." \
  -F "negative_prompt=' '" \
  -F "size=2048x2048" \
  -F "num_inference_steps=50" \
  -F "guidance_scale=4.0" \
  -F "seed=42" \
  -F "response_format=b64_json" \
  | jq -r '.data[0].b64_json' | base64 -d > output.png
```

Qwen-Image
```
VLLM_LOGGING_LEVEL=DEBUG vllm serve Qwen/Qwen-Image --omni --port 8098 --ulysses-degree 8

curl -s -X POST http://localhost:8098/v1/images/generations \
  -H "Content-Type: application/json" \
  -d '{
    "model": "Qwen/Qwen-Image",
    "prompt": "A cat holding a sign that says hello world.",
    "negative_prompt": " ",
    "size": "2048x2048",
    "num_inference_steps": 50,
    "guidance_scale": 0.0,
    "seed": 42,
    "response_format": "b64_json"
  }' | jq -r '.data[0].b64_json' | base64 -d > output.png
```

Flux2
```
VLLM_LOGGING_LEVEL=DEBUG vllm serve black-forest-labs/FLUX.2-dev \
  --omni \
  --port 8098 \
  --ulysses-degree 8

curl -s -X POST http://localhost:8098/v1/images/generations \
  -H "Content-Type: application/json" \
  -d '{
    "prompt": "A small cat",
    "size": "2048x2048",
    "num_inference_steps": 25,
    "guidance_scale": 0.0,
    "seed": 42,
    "response_format": "b64_json"
  }' \
  | jq -r '.data[0].b64_json' \
  | base64 -d > output.png
```

#### Summary
```
Model             Baseline (ms)    PR (ms)    Speedup
-----------------------------------------------------
qwen-image-edit        34316.45   34005.69     0.9%
qwen-image             18376.40   16948.19     7.8%
flux2                  12098.75   11910.14     1.6%
wan2.2                152967.75  151036.75     1.3%
```

#### Example traces
Wan2.2 Baseline
baseline

Wan2.2 PR
pr

Flux2 Baseline
flux2_traces_baseline

Flux2 PR
flux2_traces_pr

#### Example Outputs
Wan2.2 baseline

https://github.com/user-attachments/assets/7d4c737e-e736-4813-8ebb-ecd0f2de0850

Wan2.2 PR

https://github.com/user-attachments/assets/bb5100af-1d3a-4c4e-b1a2-6b1eb0b484de

### CODE REVIEW

**Review by hsliuustc0106 — COMMENTED** [review]
Reviewed.

File: vllm_omni/diffusion/compile.py [hunk 1]
```diff
@@ -27,6 +27,18 @@ def regionally_compile(model: nn.Module, *compile_args: Any, **compile_kwargs: A
         logger.warning("Regional compilation skipped because the model does not define `_repeated_blocks`.")
         return model
 
+    # Opt-in: reorder compute/comm to overlap collectives with compute.
+    if getattr(model, "_reorder_compute_comm_overlap", False):
```
[the remark this answers is not in the document]
Comment (kTorp) [reply vllm_omni/diffusion/compile.py]: Oversight on my part. Commit for Qwen-Image transformer had not been pushed before un-drafting.

**Comment by kTorp** [comment]
hi @hsliuustc0106, just wanted to check if there's any interest in this addition. I can update the PR and run any requested tests if so

Thanks!

**Comment by hsliuustc0106** [comment]
does it change to cli interface? any documentation update related to this update?
I suggest not to make it as a default.

**Comment by hsliuustc0106** [comment]
can you run the accuracy under tests/ locally and paste the results

**Comment by kTorp** [comment]
> does it change to cli interface? any documentation update related to this update? I suggest not to make it as a default.

Hi, I've updated the implementation using the `diffusion-compile-dynamic` flag as a guide and added appropriate documentation. It is no longer active by default. A/B testing still shows similar improvements. The setting is now activated with

`vllm serve <model> --omni --port 8098 --ulysses-degree 8 --diffusion_compile_reorder_comm_overlap`

Is this a better solution or do you prefer the old way?

**Comment by kTorp** [comment]
Closing to rework and update for newer release versions and be more inline with updated merge processes.
